broker/service/api/v10.py

```diff
@@ -153,3 +163,378 @@ def submission_log(submission_id):
+
+        status = "success"
+        reason = ""
+
+    return {"cluster_name": conf_name, "status": status, "reason": reason}
+
+""" Activate a cluster to be used in a Asperathos section
+
+Raises:
+    ex.BadRequestException -- Missing parameters in request
+    ex.UnauthorizedException -- Authetication problem
+
+Returns:
+    dict -- Returns a dict with the cluster_name, 
+    the status of the activation (success or failed) and a
+    reason in case of 'failed' status
+"""
+
+def activate_cluster(cluster_name, data):
+
+    if ('enable_auth' not in data):
+        API_LOG.log("Missing parameters in request")
+        raise ex.BadRequestException()
+
+    enable_auth = data['enable_auth'] 
+
+    if enable_auth:
+        if 'username' not in data or 'password' not in data:
+            API_LOG.log("Missing parameters in request")
+            raise ex.BadRequestException()
+
+        username = data['username']
+        password = data['password']
+
+        authorization = authorizer.get_authorization(api.authorization_url,
+                                                 username, password)
+                                            
+        if not authorization['success']:
+            API_LOG.log("Unauthorized request")
+            raise ex.UnauthorizedException()
+
+    try:
+        controller.active_cluster(api.controller_url, cluster_name, data)
+        monitor.active_cluster(api.monitor_url, cluster_name, data)
+        visualizer.active_cluster(api.visualizer_url, cluster_name, data)
+    except Exception:
+        API_LOG.log("Error while activating the cluster into other components")
+
+    conf_name = cluster_name
+
+    if(not os.path.isfile("./data/clusters/%s/%s" % (conf_name, conf_name))):
+        status = "failed"
+        reason = "cluster does not exists in this Asperathos section"
+    else:
+
+        with open("./data/clusters/%s/%s" % (conf_name, conf_name), 'r') as f:
+            with open(api.k8s_conf_path, 'w') as f1:
+                for line in f:
+                    f1.write(line) 
+        
+        status = "success"
+        reason = ""
+
+    return {"cluster_name": conf_name, "status": status, "reason": reason}
+
+""" Get the list of usable clusters in the Asperathos Manager instance
+
+Returns:
+    dict -- Returns a dict with the cluster_name as key and
+    the cluster config content as value.
+"""
+def get_clusters():
+
+    response = {}
+    for root, dirs, files in os.walk("./data/clusters"):
+        for name in dirs:
+            cluster_file = open("./data/clusters/%s/%s" % (name, name), mode='r')
+            cluster_config = cluster_file.read()
+            response[name] = cluster_config
+    
+    return response
+
+""" Get the current active cluster in Asperathos Manager instance
+
+Returns:
+    dict -- Returns a dict with the cluster_name as key and
+    the cluster config content as value.
+"""
+def get_activated_cluster():
```

Comment:
using a data structure as described in the previous comment would also make this method effortless.

Reply:
Modification made in commit [540b521](https://github.com/ufcg-lsd/asperathos-manager/commit/540b521953feea4e73758f6d8173a1c58e9e65be)